tv_apps/smart-home/test/unit/home_test.js

```diff
@@ -0,0 +1,382 @@
+      var testCard = document.createElement('div');
+      testCard.dataset.cardId = '123';
+
+      var testApp = new Application({});
+      testApp.launch = this.sinon.stub();
+      subject.cardManager.m_card['123'] = testApp;
+
+      subject.handleFocus(testCard);
+      subject.onEnter();
+      assert.isTrue(testApp.launch.calledOnce);
+    });
+  });
+
+  suite('onSearchBarShown >', function() {
+    setup(function() {
+      this.sinon.stub(document, 'addEventListener')
+        .withArgs('visibilitychange').callsArg(1);
+      subject.searchBar.hide = this.sinon.stub();
+    });
+
+    test('should fire search activity', function() {
+      assert.equal(MozActivity.calls.length, 0);
+      subject.onSearchBarShown();
+      assert.equal(MozActivity.calls[0].name, 'search');
+    });
+
+    test('should hide search bar after visibility has changed', function(done) {
+      subject.searchBar.hide = function() {
+        assert.isTrue(true);
+        done();
+      };
+      subject.onSearchBarShown();
+    });
+  });
+
+  suite('onSearchBarHidden >', function() {
+    test('should show search button again', function() {
+      searchButton.classList.add('hidden');
+      subject.onSearchBarHidden();
+      assert.isFalse(searchButton.classList.contains('hidden'));
+    });
+  });
+
+  suite('handleFocus >', function() {
+    test('should switch focus to scrollable target', function() {
+      var spy = this.sinon.spy(subject.cardScrollable, 'focus');
+      subject.handleFocus(subject.cardScrollable);
+      assert.isTrue(spy.calledOnce);
+    });
+
+    test('should switch focus to next focus for normal target', function() {
+      var spy = this.sinon.spy(editButton, 'focus');
+      subject.handleFocus(subject.editButton);
+      assert.isTrue(spy.calledOnce);
+    });
+
+    test('should open menu for menu group target', function() {
+      var stub = this.sinon.stub(subject, 'handleFocusMenuGroup');
+      subject.handleFocus(filterTabGroup);
+      assert.isTrue(stub.calledOnce);
+    });
+
+    test('should call checkFocusedGroup for non-menu elements', function() {
+      var stub = this.sinon.stub(subject, 'checkFocusedGroup');
+      subject.handleFocus(settingsButton);
+      assert.isTrue(stub.withArgs(settingsButton).calledOnce);
+    });
+  });
+
+  suite('checkFocusedGroup >', function() {
+    test('should close group menu when blur', function() {
+      subject.handleFocusMenuGroup(filterTabGroup);
+      var stub = this.sinon.stub(filterTabGroup, 'close');
+      subject.checkFocusedGroup();
+      assert.isTrue(stub.calledOnce);
+    });
+
+    test('should keep settings menu open when switching back from edit mode',
+    function() {
+      subject.handleFocusMenuGroup(settingsGroup);
+      var stub = this.sinon.stub(settingsGroup, 'close');
+      subject.edit.mode = 'edit';
+      subject.handleFocus(subject.cardScrollable);
+      subject.checkFocusedGroup(settingsGroup);
+
```

Comment:
If any of these invocation called `settingsGroup.close()`, we'd get `stub.called` true in the end. So the result remains the same whether we add one line here or not. But yeah It can help debugging.